libusb/hid.c

```diff
@@ -717,7 +738,8 @@ static int is_xbox360(unsigned short vendor_id, const struct libusb_interface_de
 		0x9886, /* ASTRO Gaming */
 	};
 
-	if (intf_desc->bInterfaceClass == LIBUSB_CLASS_VENDOR_SPEC &&
+	if (intf_desc &&
```

Comment:
a separate check at the beginning of the function to separate the actual logic of the implementation and the technicalities